src/services/authServices.ts

```diff
@@ -0,0 +1,122 @@
+import brcypt from 'bcrypt';
+
+import Users from '../types/users';
+import UserModal from '../Models/user';
+
+const HASH_SALT_ROUND = 10;
+
+async function getUserByLoginCred(email: string) {
+  const result = await UserModal.find({ email: email }).populate('boards', 'title').select('-__v');
+  return result;
+}
+
+async function authenticateUser(password: string, hashedPassword: string): Promise<boolean> {
+  try {
+    const result = await brcypt.compare(password, hashedPassword);
+
+    return result;
+  } catch (err) {
+    console.log(err, 'There was a error while compairing hash value');
+
+    return false;
+  }
+}
+
+async function hashPassword(password: string) {
+  const result = await brcypt.hash(password, HASH_SALT_ROUND);
+
+  return result;
+}
+
+async function registerUser(data: Users) {
+  return UserModal.insertMany({
+    firstName: data.firstName,
+    lastName: data.lastName,
+    email: data.email,
+    password: data.password,
+  });
+}
+
+export const loginService = async (data: Pick<Users, 'email' | 'password'>) => {
+  const { email, password } = data;
+
+  try {
+    const [user] = ((await getUserByLoginCred(email)) as unknown) as Users[];
+
+    if (!user) {
+      return {
+        status: 404,
```

Comment:
Use constants:
- It makes your code readable, maintainable, and reusable.

NOT_FOUND = 404
FORBIDDEN = 403

Alternatively, you can use https://www.npmjs.com/package/http-status-codes for HTTP Codes.